Export plot

William-Hill opened the pull request:
Resolves #30

William-Hill committed:
began adding export plot modal

William-Hill committed:
adding spinner

William-Hill committed:
updated reactstrap to use Spinner

William-Hill committed:
added spinner label

William-Hill committed:
plot export is functional; made UI tweaks for usability

William-Hill committed:
adding custom dimensions UI components

William-Hill committed:
added radio buttons for plot units

William-Hill committed:
raise exception when caught in inject() function

William-Hill committed:
attempting to fix plotString

William-Hill committed:
raise error when caught in inject() function

William-Hill committed:
added functionality to export plot in Postscript

William-Hill committed:
removed throwing Error as catch-all; checking if nbformat isStream

William-Hill committed:
switch ordering of template and graphics method in plotString;

William-Hill committed:
added exporting other file formats besides png in custom dimensions

downiec commented on src/cell_utils.ts:
```diff
@@ -29,10 +29,14 @@ namespace cell_utils {
       return null;
     }
     let out = cell.outputs.toJSON().pop();
+    console.log("out:", out)
```
don't need this console.log anymore, delete

William-Hill replied:
I removed this and other logging statements with a new commit.

downiec commented on src/components/VCSMenu.tsx:
```diff
@@ -337,20 +437,112 @@ export class VCSMenu extends React.Component<VCSMenuProps, VCSMenuState> {
+    console.log("width in save:", this.state.width)
+    console.log("plotUnits in save:", this.state.plotUnits)
+
+    if(fileFormat === "png"){
```
Change to: if(fileFormat == "png"), to match the others. Unless there's a reason for it to need '===' ?

William-Hill replied:
I used triple equals for strict equivalence checking

William-Hill committed:
removed logging statements; added triple equals for strict equals checking

sterlingbaldwin commented:
This is a pretty significant addition to the VCSMenu as far as lines of code goes. I think the modal should be broken out into its own component.

downiec commented:
I agree, it should be separated into a separate component.

downiec reviewed (changes requested):
As pointed out by Sterling, once the code can be made into its own component, I think it will be good to go

William-Hill committed:
extracted ExportPlotModal into its own component

William-Hill committed:
Created props object for ExportPlotModal

William-Hill commented:
Pushed comments to break the modal into its own component @downiec @sterlingbaldwin

downiec commented on src/components/VCSMenu.tsx:
```diff
@@ -1,7 +1,7 @@
 import * as React from "react";
-import { Button, Card, CardBody } from "reactstrap";
-
+import { Alert, Button, ButtonGroup, Card, CardBody, Collapse, CustomInput, Modal, ModalHeader, ModalBody, ModalFooter, Form, FormGroup, Input, Label } from "reactstrap";
```
Its good to remove any unnecessary imports and leave them only in the file they're needed. For now, I'll remove them in the next PR after this one is merged.

downiec reviewed (changes requested):
Just tested it, and I noticed that every time I click the toggle button to enter my own custom dimensions, the dimensions get injected into the notebook. I think it would be better if the injection is done after the user finalizes their decisions and clicks 'export', otherwise the notebook can quickly get spammed with injected code. Also I keep getting an error: 'Code resulted in errors. Error name: Exception. Message: Nothing to dump aborting.' so we need to get that fixed before merging.

downiec committed:
Checks whether a plot exists before allowing user option to export a plot. Updated code so that the canvas dimensions are captured in the background instead of through cell injection. Main widget now keeps track of whether a plot exists and updates based on current selected notebook. Also did some cleanup of unneeded dependencies.

downiec committed:
Minor cleanup

downiec committed:
Minor cleanup plus added fix so that when plot injection is made, the plotExist state is set to true immediately.

downiec committed:
Merged in latest next branch

sterlingbaldwin commented on src/components/ExportPlotModal.tsx:
```diff
@@ -0,0 +1,353 @@
+            </div>
+          </div>
+          <br />
+          {/* <CustomInput type="switch" id="exampleCustomSwitch" name="customSwitch" label="Capture Provenance" checked={this.state.captureProvenance} onChange={this.toggleCaptureProvenance} /> */}
```
Commented out code should be removed before its merged.

sterlingbaldwin commented on src/components/VCSMenu.tsx:
```diff
@@ -325,20 +408,23 @@ export class VCSMenu extends React.Component<VCSMenuProps, VCSMenuState> {
       }
 
-      console.log("plotString:", plotString)
+      console.log("plotString:", plotString);
```
Debug console logs should be removed before merge.

sterlingbaldwin reviewed (changes requested):
Just a couple minor issues, nothing serious.

William-Hill committed:
removed comment

William-Hill committed:
removed code for provenance capture; will create a new PR to include after backend bug is fixed

William-Hill commented:
Just removed the comment and removed the provenance capture. I'll create a separate PR for the provenance capture after the backend part is fixed @sterlingbaldwin

sterlingbaldwin commented:
I just merged the help_links PR, which includes all these changes so Im closing this.
